packages/marketplace/test/common/AccessControl.behavior.ts

```diff
@@ -0,0 +1,56 @@
+import {expect} from 'chai';
+import {deployFixtures} from '../fixtures.ts';
+import {loadFixture} from '@nomicfoundation/hardhat-network-helpers';
+import {Contract, Signer} from 'ethers';
+
+// eslint-disable-next-line mocha/no-exports
+export function checkAccessControl(
+  functionName: string[],
+  eventName: string[],
+  userContract: string[],
+  adminContract: string[],
+  role: string[]
+) {
+  describe('Access Control', function () {
+    let ExchangeContractAsUser: Contract,
+      ExchangeContractAsAdmin: Contract,
+      user: Signer,
+      DEFAULT_ADMIN_ROLE: string,
+      contractMap: {[key: string]: Contract},
+      roleMap: {[key: string]: string};
+
+    beforeEach(async function () {
+      ({
+        ExchangeContractAsAdmin,
+        ExchangeContractAsUser,
+        user,
+        DEFAULT_ADMIN_ROLE,
+      } = await loadFixture(deployFixtures));
+      contractMap = {
+        ExchangeContractAsAdmin: ExchangeContractAsAdmin,
+        ExchangeContractAsUser: ExchangeContractAsUser,
```

Comment:
can we make this behaviour generic and not specific to exchange ?

Reply:
Its not generic to Exchange and we can add more key-value pair if needed.